packages/teams-js/src/private/copilot.ts

```diff
@@ -41,12 +48,53 @@ export namespace copilot {
      *
      * @throws Error if {@linkcode app.initialize} has not successfully completed
      */
-    export function getEligibilityInfo(): AppEligibilityInformation {
+    export function getEligibilityInfo(): Promise<AppEligibilityInformation> {
       ensureInitialized(runtime);
       if (!isSupported()) {
         throw errorNotSupportedOnPlatform;
       }
-      return runtime.hostVersionsInfo!.appEligibilityInformation!;
+
+      // Return the eligibility information if it is already available
+      if (runtime.hostVersionsInfo?.appEligibilityInformation) {
+        return Promise.resolve(runtime.hostVersionsInfo!.appEligibilityInformation);
```

Comment:
If we write async in the function header, the returned values will automatically be wrapped in promise. I think it can be simplified to  `return runtime.hostVersionsInfo!.appEligibilityInformation!`